### ISSUE TITLE
Customized date range with time

### ISSUE THREAD

**Issue by mpeterka** [javamelody/javamelody#613]
There is Choice of period "Customized", which supports only date, not time - minimal period is 24 hours.

Accepting time would be nice. (Input itself accepts time of the day, but value is ignored.)
Sometimes it is useful to browse historical results more fine-grained. (For example: top slow SQL at night.)

**Comment by evernat** [comment]
Hi @mpeterka 
There was a discussion about more granularity in graphs and data views, see #327, which is closed as won't fix.
But as said in that issue, graphs can be zoomed and then zoomed again with the slider above zoomed graphs. And a check box above zoomed graphs can hide maximums to better see average values.

**Comment by mpeterka** [comment]
Thanks.

One thing is viewing graphs, other thing is viewing statistics (sql, http).

### Use case:
There was big data import to database in the morning. Customer complains about slow app. We add database table index in the afternoon.
**Question was**: _is adding this index enough? Is there any other slow SQL queries?_ 
**Javemelody cant answer this**, because there is no option to filter "Statistics sql" in parts (time intervals) "before data import - morning" / "after data import" / "after add index".

**Comment by evernat** [comment]
Yes, you can't filter statistics data hour by hour and in fact, data of http and sql events are not stored individually, because that would give a very big volume of data and a very different monitoring overhead than currently.
But you can filter and compare statistics data between yesterday and today, then you will be able to do the same for tomorrow (beside some more "real time" data in graphs and in the display of current requests / threads) : that's enough in my experience and in general.
Knowing each http and sql event is not worth it.

**Comment by bedirachit** [comment]
Hey Is there any way I can get the data on which the graph is made for Http request for the current day. Since that graph shows the hourly request data, I can really use that.

**Comment by evernat** [comment]
@bedirachit From a browser, there is currently no way to get raw data from a graph, except the last added value (for example: http://javamelody.org/demo/monitoring?part=lastValue&graph=usedMemory). From the webapp or from the server, you may use [jrobin-1.5.9.jar](https://repo1.maven.org/maven2/org/jrobin/jrobin/1.5.9/jrobin-1.5.9.jar) to read data in rrd files on disk but some code is currently needed using the jrobin classes for that.
But to make that easier, I will think of adding a page like for the last value, but for all values in a period.

**Comment by bedirachit** [comment]
@evernat Thank for your quick response.
 
What exactly is the time frame for the last added value? For Eg: I tried for httpHitsRate and it returned 0.0, Does it mean the system received 0 requests in last one minute? 

Also what I need is to get minute wise data for httpHitsRate and others. I see that you have graphs available for those, so I am sure that data is available in rdd files. I also checked the external API, but they don't return consolidated data. 

Any idea how I can achieve this?

**Comment by evernat** [comment]
@bedirachit A new value is added every minute. So the last value may not contain all the latest hits. If 0.0, yes, 0 requests were executed for the last minute when the value was added.
I have added dump as xml or txt for the next release (1.83), see [doc](https://github.com/javamelody/javamelody/wiki/ExternalAPI#dump-graph-as-xml-or-txt). A snapshot build is at http://javamelody.org/downloads/javamelody-core-1.83.0-SNAPSHOT.jar
